api/catalog/management/commands/generatewaveforms.py

```diff
@@ -0,0 +1,77 @@
+import logging
+import subprocess
+
+from catalog.api.models.audio import Audio, AudioAddOn
+from django_tqdm import BaseCommand
+
+
+def paginate_reducing_query(get_query_set, page_size=10):
+    """
+    We can't use `Paginator` because it can't handle the situation
+    where the query result changes each time a page is accessed.
+    Because the `audios` QuerySet result is naturally getting smaller
+    each time we successfully process waveforms, we can just take
+    the first ten for each "page" until the page comes back empty.
+    This should theoretically be faster/less DB latency inducing
+    anyway as we're never going to have huge OFFSET values to
+    access deep pages.
+    """
+    page = list(get_query_set()[0:page_size])
+    while len(page):
+        yield page
+        page = list(get_query_set()[0:page_size])
+
+
+class Command(BaseCommand):
+    help = "Generates waveforms for all audio records to populate the cache."
+    """
+    Note: We rely on the file download and waveform generation times
+    taking long enough to prevent us from either making too many requests
+    to the upstream provider or inserting into our database too quickly and
+    causing a slow down. In local tests and in tests run on the staging server
+    it appeared to take on average around 6 to 8 seconds for each audio file.
```

Comment:
I think it should depend on the size of the audio file. Many Freesound and Wikimedia files are very small. I sometimes see 2-3 it/s in the console, and the maximum is 11 it/s.

Reply:
Right. I think we probably should use the `limit` library anyway just in case.